cluster-autoscaler/cloudprovider/openshiftmachineapi/machineapi_controller.go

```diff
@@ -415,6 +421,12 @@ func (c *machineController) nodeGroupForNode(node *apiv1.Node) (cloudprovider.No
 		return nil, fmt.Errorf("failed to build nodegroup for node %q: %v", node.Name, err)
 	}
 
+	// We don't scale from 0 so nodes must belong to a nodegroup
+	// that has a scale size of at least 1.
+	if nodegroup.MaxSize()-nodegroup.MinSize() < 1 {
```

Comment:
Can you also add a testcase where you set both sizes to be equal and verify you get nil node group?

Reply:
Done in https://github.com/openshift/kubernetes-autoscaler/pull/42/commits/eec7f96068779625b486818609d6e3b0f761f256

Reply:
And again in https://github.com/openshift/kubernetes-autoscaler/pull/42/commits/c3dcd129b96f380dafaf05256ac34e16356adfce

Reply:
Test cases now covered in `TestControllerNodeGroupForNodeLookup()` - https://github.com/openshift/kubernetes-autoscaler/pull/42/commits/56bc385435166024ea03dd2fd020327680e96229